xarray/core/dataset.py

```diff
@@ -7479,11 +7506,20 @@ def to_dask_dataframe(
         import dask.array as da
         import dask.dataframe as dd
 
+        if not create_index and set_index:
+            raise ValueError("create_index=False is incompatible with set_index=True")
+
         ordered_dims = self._normalize_dim_order(dim_order=dim_order)
 
-        columns = list(ordered_dims)
-        columns.extend(k for k in self.coords if k not in self.dims)
-        columns.extend(self.data_vars)
+        if create_index:
+            columns = list(ordered_dims)
```

Comment:
I think we still want these dimension columns  in the dask dataframe but we don't want to set them as an index?